code/modules/awaymissions/maploader/reader.dm

```diff
@@ -304,6 +311,16 @@ var/list/map_dimension_cache = list()
 	members.Remove(members[index])
 
 	//then instance the /turf and, if multiple tiles are presents, simulates the DMM underlays piling effect (only the last turf is spawned, other ones are drawn as underlays)
+	if(overwrite) //make this come first so lighting overlays don't die
+		var/turf/T_old = locate(xcrd,ycrd,zcrd)
+		var/static/list/blacklisted_types = list(/mob/dead/observer,/mob/dview,/atom/movable/lighting_overlay)
+		var/list/truncated_contents = T_old.contents.Copy()
+		while(truncated_contents.len)
+			truncated_contents = T_old.contents.Copy()
+			for(var/type in blacklisted_types)
+				truncated_contents = prune_list_to_type(truncated_contents,type,TRUE)
+			for(var/atom/movable/thing in truncated_contents)
```

Comment:
```suggestion
			for(var/thing in truncated_contents)
```
Should be faster since it doesn't have to typecheck every element of the list

Reply:
I think I did this because it was catching turfs, but I should test it without it again.

Reply:
It shouldn't be possible for a turf to be in a turf's contents list